webrender/src/resource_cache.rs

```diff
@@ -259,10 +259,16 @@ impl ResourceCache {
 
     fn should_tile(&self, descriptor: &ImageDescriptor, data: &ImageData) -> bool {
         let limit = self.max_texture_size();
+        let size_check = descriptor.width > limit || descriptor.height > limit;
         return match data {
-            // Tiled external images are not implemented.
+            &ImageData::Raw(_) => { size_check }
+            &ImageData::Blob(_) => { size_check }
+            &ImageData::External(ExternalImageData { image_type: ExternalImageType::ExternalBuffer, .. }) => {
+                size_check
```

Comment:
nit could do `data.image_type == ExternalImageType::ExternalBuffer && size_check` instead of pattern matching